osbuild/util/bls.py

```diff
@@ -0,0 +1,35 @@
+import glob
+import os
+
+"""
+Function for appending parameters to
+Boot Loader Specification (BLS).
+"""
+
+def options_append(root_path, kernel_arguments):
+    """
+    Add kernel arguments to the Boot Loader Specification (BLS) configuration files.
+    There is unlikely to be more than one BLS config, but just in case, we'll iterate over them.
+
+    Parameters
+    ----------
+
+    root_path (str): The root path for locating BLS configuration files.
+    kernel_arguments (list): A list of kernel arguments to be added.
+
+    """
+    entries = []
+    for entry in glob.glob(f"{root_path}/loader/entries/*.conf"):
+        entries.append(entry)
+        # Read in the file and then append to the options line.
+        with open(entry, encoding="utf8") as f:
+            lines = f.read().splitlines()
+        with open(entry, "w", encoding="utf8") as f:
+            for line in lines:
+                if line.startswith('options '):
```

Comment:
no `options` line would mean there are no kernel arguments, which I think would never happen.

I don't know for sure if the behavior is defined for grub/BLS if there is no options line. Even if it didn't error you'd end up with a system that wouldn't boot.

Reply:
I checked the spec at https://uapi-group.org/specifications/specs/boot_loader_specification/ and it says `options shall contain kernel parameters to pass to the Linux kernel to spawn. This key is optional and may appear more than once in which case all specified parameters are combined in the order they are listed.` - so we need to tweak this also a little bit in a followup to be correct in the general case. It will be fine for your use-case as you (I assume) make sure the inputs are aligned in the right way (exactly one options line). I did a followup patch in https://github.com/osbuild/osbuild/compare/main...mvo5:process_json-tweaks?expand=1 that deals with the extra conditions and adds a test.

Reply:
Thanks. wow didn't know that. I bet if we exercised that functionality we'd get some undefined behavior in various tools that parse the file. 

Will review when we open the followup PR.